tests/integration/test_confirm_payments.py

```diff
@@ -2,62 +2,88 @@
 
 from django.core.management import call_command
 import pytest
+import time
 
-ETHERSCAN_API_KEY = os.environ.get('ETHERSCAN_API_KEY')
-skip_if_no_etherscan_api_key = pytest.mark.skipif(
-    ETHERSCAN_API_KEY is None,
-    reason='Etherscan api key is not set',
-)
-
-# Fake DAI deployment (LAI token)
-GOERLI_TOKEN_ADDRESS = '0x57865b333088b4369c77e11b8c0f410ca2242e09'
-GOERLI_WALLET_ADDRESS = '0xddd41b99968179d5f08b93a45eee20108c1b3f95'
+from django.test import RequestFactory
 
+from pretix_eth.models import WalletAddress
 
-@pytest.fixture
-def get_orders_and_payments(get_order_and_payment):
-    def _get_orders_and_payments(info_data_list):
-        orders = []
-        payments = []
 
-        for info_data in info_data_list:
-            order, payment = get_order_and_payment(info_data=info_data)
-
-            orders.append(order)
-            payments.append(payment)
+WEB3_PROVIDER_URI = os.environ.get('WEB3_PROVIDER_URI')
+skip_if_no_web3_provider = pytest.mark.skipif(
```

Comment:
Worth pointing out that this pattern can be a bit of a "foot gun".  If the `WEB3_PROVIDER_URL` is not set then all the tests will just silently skip.

Typically, for tests like this I've found it better to have a way to segment the tests so that the ones that *need* the provider can be explicitly run but that they will fail loudly if not properly configured.  This allows the main test suite to be run without this configuration, and then the other suite can only be run in environments where the config is known to be present.

An alternative pattern is to have an extra way to *flag* this as required so that by default it will skip, but so that you can run the tests with an explicit flag that will cause this to instead fail if misconfigured.

Reply:
I understood your point. I will look into pytest and see how it can be implemented.

Reply:
I implemented this by calling the additional check function at the beginning of each test. This function checks if the flag `--web3` is set then decides to skip or not. If it didn't skip, meaning that flag is set, then check if it is properly configured. If it wasn't properly configured, then fails with a message.

If you approve this method I will implement the same idea in #145